features/users/operations.py

```diff
@@ -110,3 +161,154 @@ def create_token(subject: Union[str, Any], expires_delta: timedelta = None, acce
     encoded_jwt = jwt.encode(to_encode, secret_key, algorithm)
 
     return encoded_jwt, token_type
+
+
+async def send_email(*, subject: str, content: str, recipient: str):
+    """
+    Send email for email confirmation or reset password
+
+    :param subject:
+    :param content:
+    :param recipient:
+    """
+
+    # TODO: setup from_email
+    message = Mail(
+        from_email='your@example.com',
+        to_emails=recipient,
+        subject=subject,
+        html_content=content
+    )
+
+    sg = SendGridAPIClient(api_key=config.grid.send_grid_api_key)
+
+    try:
+        response = sg.send(message)
+        return {"message": "Email sent successfully", "status_code": response.status_code}
+    except Exception as e:
+        return {"message": "An error occurred", "error": str(e)}
+
+
+def generate_email_confirmation_token(user: User, expiration_days: int = 7):
+    """
+    Generate email confirmation token
+
+    :param user:
+    :param expiration_days:
+    :return:
+    """
+
+    token = secrets.token_urlsafe(32)
+
+    """Calculate the expiration datetime"""
+    expiration_time = datetime.utcnow() + timedelta(days=expiration_days)
+
+    with get_session() as session:
+        token_obj = EmailConfirmationToken(
+            email_confirmation_token=token,
+            user_id=user.id,
+            expired_on=expiration_time
+        )
+        session.add(token_obj)
+        session.commit()
+
+    return token
+
+
+def get_token_from_db(token: str, token_type: str):
```

Comment:
```suggestion
def check_if_token_is_valid(token: str, token_type: str, user_id: int):
```
if exist token with token == token, token_type==token_type, user==user_id that expires in the futere